CodeModel/src/main/java/org/openzen/zenscript/codemodel/type/builtin/MapTypeSymbol.java

```diff
@@ -87,8 +88,16 @@ public ResolvedType resolve(TypeID[] typeArguments) {
 		members.method(mapper.map(type, BuiltinMethodSymbol.ASSOC_VALUES));
 		members.method(new MethodInstance(BuiltinMethodSymbol.ASSOC_HASHCODE));
 
-		members.method(mapper.map(type, BuiltinMethodSymbol.ITERATOR_ASSOC_KEYS));
-		members.method(mapper.map(type, BuiltinMethodSymbol.ITERATOR_ASSOC_KEY_VALUES));
+		members.iterator(new IteratorInstance(
+				type,
+				new TypeID[]{type.keyType},
+				mapper.map(type, BuiltinMethodSymbol.ITERATOR_ASSOC_KEYS)
+		));
+		members.iterator(new IteratorInstance(
+				type,
+				new TypeID[]{type.keyType, type.valueType},
+				mapper.map(type, BuiltinMethodSymbol.ITERATOR_ASSOC_KEY_VALUES)
+		));
```

Comment:
💭 Discussion:
One one hand I believe this is the proper fix here.

On the other hand, we might need to adapt `MemberSet$Builder#method`:
Currently, that method either puts the member into the target's staticMethods or instanceMethods.
However, for primitives we just add all BuiltinMethodSymbol s as `.method()`s, so maybe something could be registered here as a type that does not make sense?